pkg/files/file.go

```diff
@@ -225,14 +228,32 @@ func (r *File) IsTemplate() bool {
 	if r.markedTemplate != nil {
 		return *r.markedTemplate
 	}
-	return r.isTemplate()
+	return r.shouldProcess() && r.hasTemplating()
 }
 
-func (r *File) isTemplate() bool {
+func (r *File) shouldProcess() bool {
 	t := r.Type()
 	return !r.IsLibrary() && (t == TypeYAML || t == TypeText)
 }
 
+func (r *File) hasTemplating() bool {
+	data, err := r.Bytes()
+	if err != nil {
+		panic(fmt.Sprintf("Unable to get contents of file %s: %s", r.OriginalRelativePath(), err))
+	}
+	switch r.Type() {
+	case TypeYAML:
+		for _, ann := range literal.AllAnnotations {
+			if bytes.Contains(data, ann) {
```

Comment:
i think we are at the wrong abstraction level here. if somebody has a `foo: "#@overlay/match"` content this would pick this up as being a template, even though it is not.

Reply:
I blame the beer. 🍺 

Yeah, this approach was way too literal (pun inadvertent) and subpar in pretty much every measure.

I see how I got fixated on the notion that `File.IsTemplate()` is _the_ source of truth about whether a file contains templating or not. Quite a bit of complexity flowed from that myoptia.

I see now that `files` is about user input: through a combination of filenames and file marks, indicating how they _intend_ a file to be processed.

Whether a given Document Set actually contains templating is squarely in the domain of `yamltemplate`. Leveraging that fact to adjust processing (i.e. skipping processing of files that contain nothing to process) is a decision well-made at the `workspace` level.